### PR TITLE
REGR: fix read_file from url without zip extension

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Handles (part of?) https://github.com/geopandas/geopandas/issues/2908

### CODE REVIEW

File: geopandas/io/tests/test_file.py [hunk 1]
```diff
@@ -576,6 +576,13 @@ def test_read_file_remote_zipfile_url(engine):
     assert isinstance(gdf, geopandas.GeoDataFrame)
 
 
+@pytest.mark.web
+def test_read_file_remote_zipfile_url_without_extension(engine):
+    url = "https://geonode.goosocean.org/download/480"
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file.py:581]: @ocefpaf do you know if this is a stable url that we could use for testing?

Comment (jorisvandenbossche) [reply geopandas/io/tests/test_file.py:581]: Otherwise I suppose I could upload a small zip file without extension to our github repo. That's maybe better

Comment (ocefpaf) [reply geopandas/io/tests/test_file.py:581]: Stable enough and you could use pytest-vcr too. But adding it to GH is probably better.

**Comment by jorisvandenbossche** [comment]
This doesn't solve all cases (also other urls that are not "zip files without extension" are failing now), so closing in favor of https://github.com/geopandas/geopandas/pull/2914

**Pull request closed without merging**